src/hooks/useNavigate/useNavigate.ts

```diff
@@ -0,0 +1,22 @@
+import { NavigateOptions as ReactNavigateOptions, useNavigate as ReactUseNavigate } from 'react-router-dom';
+
+import { HasParams, NavigateOptions, PathParams } from 'types/params-utils';
+
+export const useNavigate = () => {
+  const navigate = ReactUseNavigate();
+
+  return <Path extends string>(
+    ...[path, options]: HasParams<Path> extends true
+      ? [path: Path, options: NavigateOptions<PathParams<Path>>]
+      : [path: Path, options?: ReactNavigateOptions]
+  ) => {
+    let parsedPath: string = path;
+    if (options && 'params' in options) {
+      Object.entries(options.params).forEach(([key, value]) => {
+        parsedPath = parsedPath.replace(`:${key}`, value as string);
+      });
+    }
+
+    navigate(parsedPath);
+  };
```

Comment:
what do you think about making this part immutable?
```suggestion
    if (options && 'params' in options) {
      return navigate(Object.entries(options.params).reduce(([key, value]) => {
  return parsedPath.replace(`:${key}`, value);
}, path))
    }

    navigate(path);
  };
```

Reply:
nice one, I think your reduce definition lacks accumulator, I've added it 🙂

Reply:
ahh right, my bad 😅